spec/requests/publishers/vacancies/job_applications/messages_spec.rb

```diff
@@ -0,0 +1,76 @@
+require "rails_helper"
+
+RSpec.describe "Publishers::Vacancies::JobApplications::Messages" do
+  let(:publisher) { create(:publisher) }
+  let(:organisation) { create(:school) }
+  let(:vacancy) { create(:vacancy, :expired, organisations: [organisation]) }
+  let(:job_application) { create(:job_application, :status_submitted, vacancy: vacancy) }
+
+  before do
+    sign_in(publisher, scope: :publisher)
+    # rubocop:disable RSpec/AnyInstance
+    allow_any_instance_of(Publishers::BaseController).to receive(:current_organisation).and_return(organisation)
+    # rubocop:enable RSpec/AnyInstance
+  end
+
+  after { sign_out publisher }
+
+  describe "POST /organisation/jobs/:vacancy_id/job_applications/:job_application_id/messages" do
```

Comment:
I'm really unsure about creating these request specs. If we have coverage for functional specs, do we need these? Controllers should not do very much, so this feels like we're 'over-testing' .